cryptoservice/crypto_service.go

```diff
@@ -82,10 +82,15 @@ func (cs *CryptoService) GetPrivateKey(keyID string) (k data.PrivateKey, role st
 	for _, ks := range cs.keyStores {
 		for _, keyPath := range keyPaths {
 			k, role, err = ks.GetKey(keyPath)
-			if err != nil {
+			if err == nil {
+				return
+			}
+			switch err.(type) {
+			case trustmanager.ErrPasswordInvalid, trustmanager.ErrAttemptsExceeded:
```

Comment:
Although I agree this is probably not intuitive to the user - we don't have any messaging about what the expected behavior is.  This is probably not the correct place to do this, but maybe we could detect whether a key exists in multiple places, and:

1. If we want the first keystore to win, say when asking for the password which keystore we're using and suggest that the user needs to do something differently if they want to use a key from a different keystore

2. List all the places the key exists, and ask the user which key they want to use so they know which credential to supply?